lenovo/thinkpad/x1-gen6/default.nix

```diff
@@ -0,0 +1,41 @@
+# pulled out all the x1 specific things
+# for nixos-hardware
+
+{ config, pkgs, ... }:
+
+{
+
+  boot.kernelModules = [ "acpi" "thinkpad-acpi" "acpi-call" ];
+  boot.extraModulePackages = [ config.boot.kernelPackages.acpi_call ];
+  boot.extraModprobeConfig = "options thinkpad_acpi experimental=1 fan_control=1";
+
+  environment.systemPackages = with pkgs; [
+    tlp
+  ];
+
+  # enable thermal trottle
+  services.thermald.enable = true;
+
+  # enable sound
+  sound.enable = true;
+  hardware.pulseaudio.enable = true;
+
+  # handle lid close hibernate, suspend, or ignore
+  systemd.extraConfig = "";
+  services.logind.extraConfig = ''
+    HandleLidSwitch=hibernate
+    LidSwitchIgnoreInhibited=yes
+  '';
+
+  # Thinkpad power services with some sample values
+  services.tlp.enable = true;
+  services.tlp.extraConfig = ''
```

Comment:
I wonder if it makes sense to put these in the default thinkpad module, since presumably they apply generally?

Reply:
This is already enabled in `common/pc/laptop` profile.

Reply:
I think there are some differences between generations. AFAIK The oldest thinkpads on the list should behave differently with `CPU_SCALING_GOVERNOR_ON_BAT=powersave` since powersave behaves differently depending on the scaling driver. Using powersave with pstate in newer generations isn't like powersave in the previous generations. 

BTW. I don't think disabling bluetooth on startup should be a default. IMHO making some tweaks here by default is okay but making tradeoffs like this is better left to be opt in.